modal/_container_entrypoint.py

```diff
@@ -534,20 +526,31 @@ def run_inputs(input_id: str, function_call_id: str, args: Any, kwargs: Any) ->
                     if not inspect.isgenerator(res):
                         raise InvalidError(f"Generator function returned value of type {type(res)}")
 
+                    # Send up to this many outputs at a time.
+                    generator_queue: asyncio.Queue[Any] = function_io_manager._queue_create(1024)
+                    generator_output_task = function_io_manager.generator_output_task(
+                        function_call_id,
+                        imp_fun.data_format,
+                        generator_queue,
+                        _future=True,  # Synchronicity magic to return a future.
+                    )
+
+                    item_count = 0
```

Comment:
I wonder if the control flow would be nicer if you have a context manager on the function io manager that handles cleanup, e.g.

```python
with function_io_manager.generator_result() as q:
    for value in res:
        q.put(value)
```
That way you can move the finalization logic to the context manager instead

This is just a thought, not at all a request! It's also entirely possible I'm misunderstanding the control flow and my suggestion is dumb

Reply:
Also with the current code, what happens if an exception is raised – does the result get sent properly to the server? I don't see a `finally:` handler or similar but maybe it's not needed

Reply:
Hmm, I tried this and it doesn't seem resolve the issue of `q` being on the synchronicity event loop and needing the hacky `function_io_manager._queue_put()` implementation. :/